### DISCUSSION TITLE
Is there any volume affinity mechanism (like there is for Pods)?

### DISCUSSION THREAD

**Discussion by Stefan-Balta** [longhorn/longhorn#5977] [Q&A]
If multiple PVCs are created with the same label(s), and each of the volumes is replicated _N_ times, is there any volume affinity mechanism to make sure that no node can have more than 1 replica of different PVs that have matching set of labels?

For example, consider Command and Query Responsibility Segregation (CQRS) pattern.
One might create 2 PVCs, one for writing and one for reading, each replicated 3 times.

Is there a way to make sure that read and write replicas never get scheduled on the same node?

**Comment by ChanYiLin** [comment]
No I don't think Longhorn has such scheduling setting
We only have setting for the replicas of the same volume
